appinventor/components/src/com/google/appinventor/components/runtime/Web.java

```diff
@@ -1431,9 +1451,8 @@ private void processResponseCookies(HttpURLConnection connection) {
     }
   }
 
-  private static String getResponseContent(HttpURLConnection connection) throws IOException {
+  private static String getResponseContent(HttpURLConnection connection, String encoding) throws IOException {
     // Use the content encoding to convert bytes to characters.
-    String encoding = connection.getContentEncoding();
```

Comment:
Hmm. We should think about what it means to specify the ResponseEncoding property. Prior to this change, the encoding specified by the server in the Content-Encoding header would be used. Effectively, this change overrides what the server specifies and uses whatever the app developer has specified. I'm not sure that I am comfortable with that. Instead, we probably want to check if encoding is null on the next line, and if so use the `responseTextEncoding` (where UTF-8 is the default).